addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/internal/KonnectedHandlerFactory.java

```diff
@@ -0,0 +1,145 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.konnected.internal;
+
+import static org.openhab.binding.konnected.KonnectedBindingConstants.*;
+
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.http.HttpServlet;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.net.HttpServiceUtil;
+import org.eclipse.smarthome.core.net.NetworkAddressService;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.openhab.binding.konnected.handler.KonnectedHandler;
+import org.openhab.binding.konnected.internal.servelet.KonnectedHTTPServlet;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link KonnectedHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Zachary Christiansen - Initial contribution
+ */
+@Component(configurationPid = "binding.konnected", service = ThingHandlerFactory.class)
+public class KonnectedHandlerFactory extends BaseThingHandlerFactory {
+    private final Logger logger = LoggerFactory.getLogger(KonnectedHandlerFactory.class);
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_MODULE);
+    private Map<ThingUID, ServiceRegistration<?>> webHookServiceRegs = new HashMap<>();
+    private HttpService httpService;
+    private String callbackUrl = null;
+    private NetworkAddressService networkAddressService;
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected void activate(ComponentContext componentContext) {
+        super.activate(componentContext);
+        Dictionary<String, Object> properties = componentContext.getProperties();
+        callbackUrl = (String) properties.get("callbackUrl");
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        KonnectedHTTPServlet servlet = registerWebHookServlet(thing);
+        return new KonnectedHandler(thing, servlet, createCallbackUrl(), createCallbackPort());
```

Comment:
I do not see any good reason why you should make the handler depend on the servlet (this creates a circular dependency).
The only use of it that I see in the code is `webHookServlet.getPath()`, but this can also easily passed as a string to the handler, if necessary.

Reply:
this dependency has been removed and the path is now passed in as a string